src/cpp/desktop/DesktopGwtCallback.cpp

```diff
@@ -244,6 +255,24 @@ QString GwtCallback::getExistingDirectory(const QString& caption,
    return createAliasedPath(result);
 }
 
+void GwtCallback::onClipboardChanged(QClipboard::Mode mode)
+{
+    // if this is a change in the selection contents, track it
+    if (mode == QClipboard::Selection)
+    {
+        QClipboard* cb = QApplication::clipboard();
+        QString text = cb->text(mode);
+
+        // when one clicks on an Ace instance, a hidden length-one selection
+        // will sneak in here. explicitly screen those out.
+        if (text == QString::fromUtf8("\x01"))
+            return;
```

Comment:
From what I understand, it's sort of a leaky implementation detail on the Ace side -- there are some comments from https://github.com/ajaxorg/ace/issues/2124, and the Ace maintainer indicates:

> this is a duplicate of #175, it needs rewriting textinput.js to not use PLACEHOLDER which is tricky to do

Behind the scenes, Ace injects the `\x01` into a hidden selection when an Ace instance is activated, and the browser / web engine treats that as a real selection and so overrides the global selection clipboard item. I don't entirely understand the technical reasons why this is done, but I imagine it's done with good reason (perhaps the only way of achieving some behavior in a consistent, cross-platform way?)